lib/day_4/board.ex

```diff
@@ -17,4 +17,62 @@ defmodule Board do
         end)
     }
   end
+
+  @spec score_of_the_winning_board({number, atom | %{:rows => any, optional(any) => any}}) ::
+          number
+  def score_of_the_winning_board({number, board}) do
+    board.rows
+    |> Enum.map(fn row -> Tuple.to_list(row.numbers) -- Tuple.to_list(row.mark) end)
+    |> List.flatten()
+    |> Enum.sum()
+    |> Kernel.*(number)
+  end
```

Comment:
it becomes more like `Board.score(board)`

number is something external to this function